state/txindex/kv/kv.go

```diff
@@ -40,6 +47,138 @@ type TxIndex struct {
 	log log.Logger
 }
 
+func (txi *TxIndex) Prune(retainHeight int64) (int64, int64, error) {
+	// Returns numPruned, newRetainHeight, err
+	// numPruned: the number of heights pruned. E.x. if heights {1, 3, 7} were pruned, numPruned == 3
+	// newRetainHeight: new retain height after pruning
+	// err: error
+
+	lastRetainHeight, err := txi.getIndexerRetainHeight()
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to look up last block indexer retain height: %w", err)
+	}
+	if lastRetainHeight == 0 {
+		lastRetainHeight = 1
+	}
+
+	ctx := context.Background()
+	results, err := txi.Search(ctx, query.MustCompile(
+		fmt.Sprintf("tx.height < %d AND tx.height >= %d", retainHeight, lastRetainHeight)))
+	if err != nil {
```

Comment:
Question to @jmalicevic. Are we sure we're not gonna hit any error in line 66? (it's a "search" query so I imagine a lot of code is gonna be hit there)

Reply:
If all goes well there should not be an error, just an empty result returned. An error should happen only in exceptional cases... The only thing I am not 100% sure of is whether we should panic here as this is not critical consensus code , so maybe it is better to log an error?

Reply:
IMO, better return an error, if possible

Reply:
Yup, its fixed now.